pkg/driftctl_test.go

```diff
@@ -0,0 +1,287 @@
+				})
+			},
+		},
+		{
+			name: "we should have changes of deleted field",
+			stateResources: []resource.Resource{
+				testresource.FakeResource{
+					Id: "fake",
+					Tags: map[string]string{
+						"tag1": "deleted",
+					},
+				},
+			},
+			remoteResources: []resource.Resource{
+				testresource.FakeResource{
+					Id: "fake",
+				},
+			},
+			assert: func(result *test.ScanResult, err error) {
+				result.AssertManagedCount(1)
+				result.AssertResourceHasDrift("fake", "FakeResource", analyser.Change{
+					Change: diff.Change{
+						Type: diff.DELETE,
+						Path: []string{"Tags", "tag1"},
+						From: "deleted",
+						To:   nil,
+					},
+					Computed: false,
+				})
+			},
+		},
+		{
+			name: "we should have changes of added field",
+			stateResources: []resource.Resource{
+				testresource.FakeResource{
+					Id: "fake",
+				},
+			},
+			remoteResources: []resource.Resource{
+				testresource.FakeResource{
+					Id: "fake",
+					Tags: map[string]string{
+						"tag1": "added",
+					},
+				},
+			},
+			assert: func(result *test.ScanResult, err error) {
+				result.AssertManagedCount(1)
+				result.AssertResourceHasDrift("fake", "FakeResource", analyser.Change{
+					Change: diff.Change{
+						Type: diff.CREATE,
+						Path: []string{"Tags", "tag1"},
+						From: nil,
+						To:   "added",
+					},
+					Computed: false,
+				})
+			},
+		},
+	}
+
+	runTest(t, cases)
+}
+
+func TestDriftctlRun_BasicFilter(t *testing.T) {
+	cases := TestCases{
+		{
+			name:           "test filtering on Type",
+			stateResources: []resource.Resource{},
+			remoteResources: []resource.Resource{
+				testresource.FakeResource{
+					Id:   "res1",
+					Type: "not-filtered",
+				},
+				testresource.FakeResource{
+					Id:   "res2",
+					Type: "filtered",
+				},
+			},
+			filter: "Type=='filtered'",
+			assert: func(result *test.ScanResult, err error) {
+				result.AssertUnmanagedCount(1)
+				result.AssertResourceUnmanaged("res2", "filtered")
+			},
+		},
+		{
+			name:           "test filtering on Id",
+			stateResources: []resource.Resource{},
+			remoteResources: []resource.Resource{
+				testresource.FakeResource{
+					Id:   "res1",
+					Type: "not-filtered",
+				},
+				testresource.FakeResource{
+					Id:   "res2",
+					Type: "filtered",
+				},
+			},
+			filter: "Id=='res2'",
+			assert: func(result *test.ScanResult, err error) {
+				result.AssertUnmanagedCount(1)
+				result.AssertResourceUnmanaged("res2", "filtered")
+			},
+		},
+		{
+			name:           "test filtering on attribute",
+			stateResources: []resource.Resource{},
+			remoteResources: []resource.Resource{
+				testresource.FakeResource{
+					Id:   "res1",
+					Type: "filtered",
+					CtyVal: func() *cty.Value {
+						data := struct {
+							TestField string `cty:"test_field"`
+						}{
+							TestField: "value to filter on",
+						}
+						ty, _ := gocty.ImpliedType(data)
+						val, _ := gocty.ToCtyValue(data, ty)
+						return &val
+					}(),
```

Comment:
You could have written the CtyVal like we did in the filter_engine tests:

```
CtyVal: func() *cty.Value {
  v := cty.ObjectVal(map[string]cty.Value{
    "test_field": cty.StringVal("value to filter on"),
  })
  return &v
}(),
```

Reply:
Obviously 🤦🏻 This is waaaay more readable thanks 👍🏻